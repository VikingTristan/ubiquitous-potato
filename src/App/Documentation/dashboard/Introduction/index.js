import React from "react";
import PrismCode from "react-prism";

import { ComponentPreview, Property, DocContainer } from "@docutils";

const BASENAME = process.env.basename;
const brand = process.env.brand;
const scriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg.js`;
const dashboardScriptUrl = `https://design.${brand}.com${BASENAME}scripts/dg-dashboard.js`;
const styleUrl = `https://design.${brand}.com${BASENAME}styles/dg-style.css`;
const dashboardStyleUrl = `https://design.${brand}.com${BASENAME}styles/dg-dashboard.css`;

const QuickStart = () => (
    <>
        <h2 id="quick-start">Quick start</h2>
        <p>To quickly add the dashboard theme to your project, include the snippets below in your project.</p>
    </>
);

const Css = () => (
    <>
        <h2 id="css">CSS</h2>
        <p>Copy-paste the stylesheet <PrismCode className="language-html">{"<link>"}</PrismCode> into your <PrismCode className="language-html">{"<head>"}</PrismCode> before all other stylesheets to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={styleUrl} />
            <link rel="stylesheet" href={dashboardStyleUrl} />
        </ComponentPreview>
    </>
);

const JavaScript = () => (
    <>
        <h2 id="javascript">JavaScript</h2>
        <p>The dashboard theme requires a separate script to run properly. So add the dashboard script below the <Property value="scripts" /> at the bottom of your <PrismCode className="language-html">{"<body>"}</PrismCode> tag like this:</p>
        <ComponentPreview language="html" codeFigure>
            <script src={scriptUrl}></script>
            <script src={dashboardScriptUrl}></script>
        </ComponentPreview>
    </>
);

const Introduction = () => (
    <DocContainer>
        <p className="lead">Get started with the dashboard theme.</p>
        <QuickStart />
        {/* <Css /> */}
        <JavaScript />
    </DocContainer>
);

export default Introduction;

/* For testing */
export { QuickStart, Css, JavaScript };
