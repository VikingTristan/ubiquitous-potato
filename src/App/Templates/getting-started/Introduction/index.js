import React from "react";

import { DocContainer } from "@docutils";

const brandTitle = process.env.brandTitle;

const Introduction = () => (
    <DocContainer docToc>
        <p className="lead">Getting started with the {brandTitle} Templates. Templates, usage description as well as tips and tricks can be found here.</p>
    </DocContainer>
);

export default Introduction;
