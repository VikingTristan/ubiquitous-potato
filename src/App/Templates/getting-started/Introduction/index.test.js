import React from "react";
import { shallow } from "enzyme";

import Introduction from "./index";

describe("Templates: Introduction", () => {
    it("is defined", () => {
        expect(Introduction).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Introduction />);

        expect(wrapper).toMatchSnapshot();
    });
});
