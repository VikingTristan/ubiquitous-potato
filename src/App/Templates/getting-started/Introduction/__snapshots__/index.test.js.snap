// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates: Introduction renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Getting started with the 
    TESTBRAND
     Templates. Templates, usage description as well as tips and tricks can be found here.
  </p>
</DocContainer>
`;
